### PR TITLE
Add support to set selection_method in GroupMod

### PR DESCRIPTION
**Pull request by wenyingd**
Add support for setting selection_method, selection_method_param and fields in a GroupMod message, the supported selection_method including "hash" and "dp_hash". When using "dp_hash", fields is supposed to be empty. "fields" could use a mask value or not, all bits in the value should be 1 if no mask is specified.

Fix: #42

### CODE REVIEW

**Comment by wenyingd** [comment]
I am a bit confused if we should bump up the version to 0.12.0 rather than 0.11.1 for this change, any idea @tnqn @antoninbas ?

File: openflow15/group.go [hunk 1]
```diff
@@ -433,3 +433,110 @@ func (prop *GroupBucketPropWatch) UnmarshalBinary(data []byte) (err error) {
...
+
+type NTRSelectionMethod struct {
+	Type             uint16
+	Length           uint16
+	ExperimenterID   uint32
+	ExperimenterType uint32
+	// padding with 4 bytes.
+	SelectionMethod [16]byte
+	SelectionParam  uint64
+	// Note, a valid field is supposed to configure with "HasMask==false", and its Value is to apply
```
Comment (tnqn) [inline openflow15/group.go:456]: what does it mean "HasMask==false"? We can't use masked field here?

Comment (wenyingd) [reply openflow15/group.go:456]: It means when we construct a MatchField object, the "HasMask" property must be set as false. So we only set "MatchField.Value", but not set "MatchField.Mask". The specified value is actually used to mask on the the given field in a packet.

**Comment by antoninbas** [comment]
> I am a bit confused if we should bump up the version to 0.12.0 rather than 0.11.1 for this change, any idea @tnqn @antoninbas ?

Maybe 0.12.0 since you are adding a new public API?

**Comment by wenyingd** [comment]
> Maybe 0.12.0 since you are adding a new public API?

Got it, updated.

**Comment by wenyingd** [comment]
@tnqn Could you help review this change?

**Review by tnqn — COMMENTED** [review]
LGTM, will update here once I verify the patch with my PR.

File: openflow15/group.go [hunk 2]
```diff
@@ -433,3 +433,110 @@ func (prop *GroupBucketPropWatch) UnmarshalBinary(data []byte) (err error) {
...
+type NTRSelectionMethod struct {
+	Type             uint16
+	Length           uint16
+	ExperimenterID   uint32
+	ExperimenterType uint32
+	// padding with 4 bytes.
+	SelectionMethod [16]byte
+	SelectionParam  uint64
+	// Note, a valid field is supposed to configure with MatchField.HasMask=false, and OVS will apply its
+	// Value to the packet field as a mask.
```
Comment (hongliangl) [inline openflow15/group.go:457]: Was the comment copied from the original C code?

Comment (wenyingd) [reply openflow15/group.go:457]: The requirement is from OVS code, but the sentence is not directly copied from source code, "MatchField", "HasMask" are struct or property defined in libOpenflow

Comment (hongliangl) [reply openflow15/group.go:457]: How about adding single quote to surround them?

Comment (wenyingd) [reply openflow15/group.go:457]: updated.

**Review by hongliangl — APPROVED** [review]

**Review by tnqn — APPROVED** [review]
LGTM

**Pull request merged**